core/domain/feedback_services.py

```diff
@@ -508,7 +514,13 @@ def get_thread_summaries(user_id, full_thread_ids):
             second_last_message_read = (
                 last_two_messages[index][1].message_id
                 in feedback_thread_user_models[index].message_ids_read_by_user)
-            author_second_last_message = last_two_messages[index][1].author_id
+            author_second_last_message = user_services.get_username(
+                last_two_messages[index][1].author_id)
+            if not last_message_read or not second_last_message_read:
```

Comment:
This slightly confuses me. In what case would the last message have been read, but the second last message be not read? If that case is impossible then this part could presumably be pulled out of both the if and else clauses into the main execution flow.